lib/evmone/eof.cpp

```diff
@@ -46,20 +60,30 @@ std::pair<EOFSectionHeaders, EOFValidationError> validate_eof_headers(bytes_view
             switch (section_id)
             {
             case TERMINATOR:
-                if (section_headers[CODE_SECTION] == 0)
+                if (section_headers[CODE_SECTION].empty())
                     return {{}, EOFValidationError::code_section_missing};
+                if (section_headers[CODE_SECTION].size() > 1 &&
+                    section_headers[TYPE_SECTION].empty())
+                    return {{}, EOFValidationError::mandatory_type_section_missing};
                 state = State::terminated;
                 break;
+            case TYPE_SECTION:
+                if (!section_headers[TYPE_SECTION].empty())
+                    return {{}, EOFValidationError::multiple_type_sections};
+                if (!section_headers[CODE_SECTION].empty())
+                    return {{}, EOFValidationError::code_section_before_type_section};
+                state = State::section_size;
+                break;
             case DATA_SECTION:
-                if (section_headers[CODE_SECTION] == 0)
```

Comment:
I think you can't remove this from here?

Reply:
It's replaced by another check in `case CODE_SECTION:`
```
if (!section_headers[DATA_SECTION].empty())
    return {{}, EOFValidationError::data_section_before_code_section};
```
(because now invalid case can be like code-data-code)

plus the old check in `case TERMINATOR` guarantees that at least one code section is present.